### PR TITLE
cheinstallation enhancements

### PR DESCRIPTION
**Pull request by nurali-techie**
Fix: https://jira.coreos.com/browse/CRT-343

- support `make up-local` with CRC
- support `make test-e2e` with CRC

### CODE REVIEW

File: test/e2e/toolchain_test.go [hunk 1]
```diff
@@ -65,41 +69,77 @@ func TestToolchain(t *testing.T) {
...
+			HasLabels(toolchain.Labels())
+
+		AssertThatOperatorGroup(t, cheOg.Namespace, cheOg.Name, f.Client).
+			Exists().
+			HasSize(1).
+			HasSpec(cheOg.Spec)
+
+		AssertThatSubscription(t, cheSub.Namespace, cheSub.Name, f.Client).
+			Exists().
+			HasSpec(cheSub.Spec)
```
Comment (MatousJobanek) [inline test/e2e/toolchain_test.go]: I believe that these assertions could be extracted to separated function that would be called at the end of every subtest.

**Comment by nurali-techie** [comment]
> Really good job

To @dipak-pawar , as he did this code and I took over from him.

File: pkg/controller/cheinstallation/cheinstallation_controller.go [hunk 2]
```diff
@@ -84,110 +101,108 @@ func (r *ReconcileCheInstallation) Reconcile(request reconcile.Request) (reconci
...
+			return true, nil
 		}
-		return true, nil
+		return false, err
 	}
 
-	return nsCreated, err
+	// To handle if namespace is deleted by user and it's in terminating state
+	if ns.Status.Phase == v1.NamespaceTerminating {
+		return false, errs.Errorf("namespace %s is not in active state", ns.Name)
```
[the remark this answers is not in the document]
Comment (nurali-techie) [reply pkg/controller/cheinstallation/cheinstallation_controller.go]: Done https://github.com/codeready-toolchain/toolchain-operator/pull/14/commits/5f1514aa59b436829a7b4b4537b892aa33afeef5

File: test/e2e/toolchain_test.go (hunk 1 above)
Comment (nurali-techie) [reply test/e2e/toolchain_test.go]: Done https://github.com/codeready-toolchain/toolchain-operator/pull/14/commits/5f1514aa59b436829a7b4b4537b892aa33afeef5

File: pkg/controller/cheinstallation/cheinstallation_controller_test.go [hunk 3]
```diff
@@ -30,101 +30,84 @@ import (
 )
 
 func TestCheInstallationController(t *testing.T) {
-	t.Run("should reconcile with che installation", func(t *testing.T) {
+	t.Run("should reconcile with che installation and create che ns", func(t *testing.T) {
```
Comment (xcoulon) [inline pkg/controller/cheinstallation/cheinstallation_controller_test.go]: the 3 subtests below depend on each other, right?

Comment (nurali-techie) [reply pkg/controller/cheinstallation/cheinstallation_controller_test.go]: @xcoulon All test case are independent due to the fact that .. each test case create new_instance of che_installation and each test case call `Reconcile()`  only one and verify the result.  So to ans your que, No sub-tests are not dependent on each other.

**Pull request merged**